packages/smooth_app/lib/pages/home_page.dart

```diff
@@ -39,6 +42,23 @@ class _HomePageState extends State<HomePage> {
   DaoProductList _daoProductList;
   DaoProduct _daoProduct;
 
+  final TextEditingController _searchController = TextEditingController();
+
+  bool _visibleCloseButton = false;
```

Comment:
I just read your code, I don't play it, but I think a close button is a good idea.